com-api-example/src/main.rs

```diff
@@ -12,3 +12,65 @@
 fn main() {
     println!("Hello, world!");
 }
+
+#[cfg(test)]
+mod test {
+    use com_api::{
+        Builder, ConsumerDescriptor, InstanceSpecifier, Producer, SampleMaybeUninit, SampleMut,
+        ServiceDiscovery, Subscriber, Subscription,
+    };
+    use com_api_sample_gen::{Tire, VehicleInterface};
+    use std::collections::VecDeque;
+
+    #[test]
+    fn create_producer() {
+        // Factory
+        let runtime_builder = com_api_sample_runtime::RuntimeBuilderImpl::new();
+        let runtime = runtime_builder.build().unwrap();
+        let producer_builder =
+            runtime.create_provided_service::<VehicleInterface>(InstanceSpecifier {});
+        let producer = producer_builder.build().unwrap();
+        let offered_producer = producer.offer().unwrap();
+
+        // Business logic
+        let uninit_sample = offered_producer.left_tire.allocate().unwrap();
+        let sample = uninit_sample.write(Tire {});
+        sample.send().unwrap();
+    }
+
+    #[test]
+    fn create_consumer() {
+        // Create runtime
+        let runtime_builder = com_api_sample_runtime::RuntimeBuilderImpl::new();
+        let runtime = runtime_builder.build().unwrap();
+
+        // Create service discovery
+        let consumer_discovery = runtime.find_instance::<VehicleInterface>(InstanceSpecifier {});
```

Comment:
Let's align this with the C++ API. Also conceptually your are searching for a service here with restrictions on the instances of that service. With that change the `get_available_instances()` call is more logical. 
```suggestion
        let consumer_discovery = runtime.find_service::<VehicleInterface>(InstanceSpecifier {});
```

Reply:
Incorporated